### ISSUE TITLE
False negative when running `pre-commit run --all-files`

### ISSUE THREAD

**Issue by mscheper** [espressif/check-copyright#18]
### Checklist

- [x] Checked the issue tracker for similar issues to ensure this is not a duplicate
- [x] Read the documentation to confirm the issue is not addressed there and your configuration is set correctly
- [x] Tested with the latest version to ensure the issue hasn't been fixed

### How often does this bug occurs?

always

### Expected behavior

For it to fail, because some of my flies still have last year's copyright notice.

### Actual behavior (suspected bug)

It passed.

### Error logs or terminal output

```shell
$ git status
[...]
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        [...]
	modified:   .pre-commit-config.yaml
	new file:   copyright.txt

$ pre-commit run --all-files
[...]
copyright-notice..............................................................................Passed
```

### Steps to reproduce the behavior

.pre-commit-config.yaml:
```yaml
default_language_version:
    python: python3.12

repos:
    # ...
    - repo: https://github.com/leoll2/copyright_notice_precommit
      rev: 0.1.1
      hooks:
          - id: copyright-notice
            args: [--notice=copyright.txt]
````

copyright.txt (in my repo root):
```
Copyright ⓒ 2025 Michael Scheper
```

helloworld/views.py (not staged, which is why I used `run --all-files`):
```python
# vim: set fileencoding=utf-8
# Copyright ⓒ 2024 Michael Scheper

from django.http import HttpResponse

def helloworld_page(request):
    return HttpResponse("Hello, world! This will be awesome!")
```

### Project release version

0.1.1

### System architecture

Intel/AMD 64-bit (modern PC, older Mac)

### Operating system

Linux

### Operating system version

Mint 21.3

### Shell

Bash

### Additional context

Other, previous hooks passed and failed as expected.

**Comment by fhrbata** [comment]
Hello @mscheper ,

shouldn't this be reported to the https://github.com/leoll2/copyright_notice_precommit project or am I missing something?

Thank you

**Comment by mscheper** [comment]
> shouldn't this be reported to the https://github.com/leoll2/copyright_notice_precommit project or am I missing something?

Apologies… I looked for alternatives when it wasn't working and I suppose I clicked in the wrong tab. 😏
